general/sonar-integration.md

```diff
@@ -0,0 +1,32 @@
+# Sonarcloud
+
+*Currently, this readme is confirmed to work for SonarCloud. The steps in this guide
+might also be relevant for other [SonarSource](https://www.sonarsource.com) products,
+such as SonarQube, but this has not been verified at this time*
+
+[Sonarcloud](https://www.sonarsource.com/products/sonarcloud/) is a service which allows
+you to scan your code for issues, and gives an online dashboard allowing you to see which
+issues exist within the code. This includes any warnings generated by analyzers. In addition,
+it has integration with DevOps platforms, which can be set up to automatically add comments
+on pull requests on any new warnings that the change has made, which can help in improving
+code quality.
+
+## The issue with this analyzer and Sonarcloud
+
+Sonarcloud's analyzer software can run while a build is being made, for example as part
+of a Pull Request check. However, Sonarcloud, by default, does not pick up any warnings
+generated by ??proj files. This means that any warnings generated by this analyzer will
+not generate comments on pull requests, which could result in a situation where they are
+not noticed and solved by developers.
+
+## Sonarcloud integration
+There is currently no 'elegant' solution to have the warnings show up on SonarCloud,
+however, there is a workaround. By adding the ??proj file as content, SonarCloud will
+recognise it's warnings and pick them up, same as any warnings in the code.
+
+For Example:
+``` XML
+  <ItemGroup>
+    <Content Include="$(MSBuildProjectFile)" CopyToOutputDirectory="Never"/>
```

Comment:
This wil only add the project file, not the props files.

Beter might be: 
``` XML
  <Content Include="{path to props}" CopyToOutputDirectory="Never"/>
  <Content Include="*.??proj" CopyToOutputDirectory="Never"/>
 ```
 
 But I'm not sure how exactly yet.

Reply:
I'm not sure adding the props file separately is necessary. I'll do a test tomorrow to see whether just adding the csproj files also picks up the props files (as the warnings are generated in the analysis of the csproj files, not the props files themselves)

Reply:
I guess not. Only the files in the content are included from what I understood.

Reply:
You were correct, appended the readme to reflect that.